src/website/app/demos/Table/examples/selectableControlled.js

```diff
@@ -0,0 +1,107 @@
+/* @flow */
+import { Component } from 'react';
+import Button from '../../../../../library/Button';
+import Flex, { FlexItem } from '../../../../../library/Flex';
+import Table from '../../../../../library/Table';
+import data from '../shared/data';
+
+export default {
+  id: 'selectable-controlled',
+  title: 'Selectable Rows - Controlled',
+  description: `Table controls its own selected rows state by default. It can
+optionally be managed by the application as a controlled component via the
+control prop, \`selectedRows\`.`,
+  scope: { Button, Component, Table, Flex, FlexItem, data },
+  source: `
+    () => {
+      const evenRows = data.filter((row, index) => (index + 1)%2 === 0);
+      const oddRows = data.filter((row, index) => (index + 1)%2 !== 0);
+
+      const columns = [
+        { content: 'Fruits', key: 'Fruits', sortable: true },
+        { content: 'Vegetables', key: 'Vegetables', sortable: true },
+        { content: 'Grains', key: 'Grains', sortable: true },
+        { content: 'Dairy', key: 'Dairy', sortable: true },
+        { content: 'Protein', key: 'Protein' }
+      ];
+
+      const initialState = {
+        selected: []
+      };
+
+      class MyTable extends Component {
+        constructor(props) {
+          super(props);
+
+          this.state = initialState;
+
+          this.handleToggleRow = this.handleToggleRow.bind(this);
+          this.handleToggleAllRows = this.handleToggleAllRows.bind(this);
+          this.selectEvenRows = this.selectEvenRows.bind(this);
+          this.selectOddRows = this.selectOddRows.bind(this);
+          this.reset = this.reset.bind(this);
+        }
+
+        handleToggleRow(row) {
+          this.setState((prevState) => {
+            const selected = prevState.selected.slice(0);
+            const index = selected.indexOf(row);
+            const hasRow = index !== -1;
+            hasRow ? selected.splice(index, 1) : selected.push(row);
+            return { selected };
+          });
+        }
+
+        handleToggleAllRows(rows) {
+          this.setState({ selected: rows });
+        }
```

Comment:
When I click a 'select even rows' & then click the indeterminate-state checkbox, instead of selecting no rows (which I believe is expected behavior), it selects all rows. At first glance, I wasn't able to discern whether that's a side-effect of this function (or if it even matters at all since this is a controlled example), but if that's the result of a built-in functionality, may be worth a thought

Reply:
Good catch; thanks. Addressed this with https://github.com/mineral-ui/mineral-ui/pull/737/commits/0b006eb4a1cb455ea8a72df8b58b6c792cfd1e5d, but you should take a look at it @brentertz (which is why it's unsquashed; details in the commit message).